plugin-gestures/src/test/java/com/mapbox/maps/plugin/gestures/GesturesPluginTest.kt

```diff
@@ -862,6 +884,202 @@ class GesturesPluginTest {
+
+    presenter = GesturesPluginImpl(context, attrs, mockk(relaxed = true))
+    presenter.bind(context, gesturesManager, attrs, 1f)
+    presenter.onDelegateProvider(mapDelegateProvider)
+    presenter.initialize()
+  }
+
+  @Test
+  fun testIsPointAboveHorizon() {
+    assertEquals(expectedResult, presenter.isPointAboveHorizon(testScreenCoordinate))
+  }
+
+  @Test
+  fun testMoveEvent() {
+    every { mapCameraManagerDelegate.cameraState } returns CameraState(
+      Point.fromLngLat(0.0, 0.0),
+      EdgeInsets(0.0, 0.0, 0.0, 0.0),
+      0.0,
+      0.0,
+      0.0
+    )
+    every {
+      mapCameraManagerDelegate.getDragCameraOptions(
+        any(),
+        any()
+      )
+    } returns CameraOptions.Builder().center(Point.fromLngLat(0.0, 0.0)).build()
+
+    val moveGestureDetector = mockk<MoveGestureDetector>()
+    every {
+      moveGestureDetector.focalPoint
+    } returns PointF(testScreenCoordinate.x.toFloat(), testScreenCoordinate.y.toFloat())
+    every { moveGestureDetector.pointersCount } returns 1
+    presenter.handleMove(moveGestureDetector, 50.0f, 50.0f)
+    verify(exactly = if (expectedResult) 0 else 1) {
+      mapCameraManagerDelegate.dragStart(any())
+    }
+  }
+
+  @Test
+  fun testFlingEvent() {
+    val motionEvent1 = mockk<MotionEvent>()
+    val motionEvent2 = mockk<MotionEvent>()
+    every { motionEvent1.x } returns testScreenCoordinate.x.toFloat()
+    every { motionEvent1.y } returns testScreenCoordinate.y.toFloat()
+    every { motionEvent2.x } returns testScreenCoordinate.x.toFloat()
+    every { motionEvent2.y } returns testScreenCoordinate.y.toFloat()
+    every { mapCameraManagerDelegate.cameraState } returns CameraState(
+      Point.fromLngLat(0.0, 0.0),
+      EdgeInsets(0.0, 0.0, 0.0, 0.0),
+      0.0,
+      0.0,
+      0.0
+    )
+    every {
+      mapCameraManagerDelegate.getDragCameraOptions(
+        any(),
+        any()
+      )
+    } returns CameraOptions.Builder().center(Point.fromLngLat(0.0, 0.0)).build()
+
+    val listener: OnFlingListener = mockk(relaxed = true)
+    presenter.addOnFlingListener(listener)
+    presenter.handleFlingEvent(motionEvent1, motionEvent2, 1000.0f, 1000.0f)
+    verify(exactly = if (expectedResult) 0 else 1) { listener.onFling() }
+    verify(exactly = if (expectedResult) 0 else 1) {
+      cameraAnimationsPlugin.easeTo(
+        any(),
+        any()
+      )
+    }
+  }
+
+  @After
+  fun cleanup() {
+    unmockkAll()
+  }
+
+  private companion object {
+    @JvmStatic
+    @ParameterizedRobolectricTestRunner.Parameters(name = "IsPointAboveHorizon using {0} projection at {1}, with MapSize {2}, testPixelForCoordinate {3} should be {4}")
+    fun data() = listOf(
+      arrayOf(
+        MapProjection.Globe,
+        ScreenCoordinate(0.0, 0.0),
+        Size(100f, 100f),
+        ScreenCoordinate(0.0, 0.0),
+        false
+      ),
+      arrayOf(
+        MapProjection.Mercator,
+        ScreenCoordinate(0.0, 0.0),
```

Comment:
also would be nice to set first coordinate of `ScreenCoordinate` bigger than `0`

Reply:
any reason to do it that way? we have tests below for ScreenCoordinates bigger than 0.